src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Check.java

```diff
@@ -461,7 +461,8 @@ boolean checkUniqueClassName(DiagnosticPosition pos, Name name, Scope s) {
             }
         }
         for (Symbol sym = s.owner; sym != null; sym = sym.owner) {
-            if (sym.kind == TYP && sym.name == name && sym.name != names.error) {
+            if (sym.kind == TYP && sym.name == name && sym.name != names.error &&
```

Comment:
Looks good!

Reply:
I recommend adding a test for this (if not already done) - checking cases like:
```
// A.java
void m() { }
class A {}
```

but also where A is nested deeper:

```
// A.java
void m() { }
class B {
   class A { }
}
```

And then also check that the test still works for other classes defined in the implicit unit:

```
// A.java
void m() { }
class B {
   class B { } //error
}
```
```

Reply:
Added